rpcserver.go

```diff
@@ -510,17 +510,47 @@ func (s *rpcServer) InitAccount(ctx context.Context,
 			"must be specified")
 	}
 
-	// Determine the desired transaction fee.
-	confTarget := req.GetConfTarget()
-	if confTarget < 1 {
-		return nil, fmt.Errorf("confirmation target must be " +
-			"greater than 0")
+	var feeRate chainfee.SatPerKWeight
+
+	switch {
+	case req.GetFeeRateSatPerKw() != 0:
+		feeRate = chainfee.SatPerKWeight(req.GetFeeRateSatPerKw())
+	case req.GetConfTarget() > 0:
+		// Determine the desired transaction fee.
+		confTarget := req.GetConfTarget()
+
+		resp, err := s.QuoteAccount(context.Background(),
+			&poolrpc.QuoteAccountRequest{
+				AccountValue: req.AccountValue,
+				Fees: &poolrpc.QuoteAccountRequest_ConfTarget{
+					ConfTarget: confTarget,
+				},
+			},
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to estimate on-chain fees: "+
+				"%v", err)
+		}
+
+		feeRate = chainfee.SatPerKWeight(resp.GetMinerFeeRateSatPerKw())
+		value := btcutil.Amount(req.AccountValue)
+
+		log.Infof("Estimated total chain fee of %v for new account with "+
+			"value=%v, conf_target=%v", feeRate, value, confTarget)
+	default:
+		return nil, fmt.Errorf("either sat/vbyte or confirmation target " +
+			"must be specified")
+	}
+
+	if feeRate < chainfee.FeePerKwFloor {
+		return nil, fmt.Errorf("fee rate of %d sat/kw is too low, "+
```

Comment:
I think we can just set the fee rate to the fee flow here, just in case the fee estimation ever returns a value too small.

Reply:
I added a check to ensure that if the fee estimation is ever too small we set ti to the FeePerKwFloor value

Reply:
Hmm, now we have the same check twice with different behavior. I think we should either remove this check down here or move the one above to here (meaning: replace the error with just setting it to the fee floor).

Reply:
This one is tricky because I would expect the caller to receive an error if they set a too low fee but in the case of specifying a `conf_target` we ensure that the user does not get that error (ensuring that we set it to,at least the minimum valid value)